### PR TITLE
Echarlie donation improvements

### LINKED ISSUE #12: Premiums should be disabled when not in radiothon mode
this prevents non-radiothon donors from filling out extraneous fields and from thinking they'll get premiums.

### LINKED ISSUE #15: /admin/donation should allow editing of fields
mostly date paid, ship date (also for pickup date, eventually), and comments.

### PR DESCRIPTION
**Pull request by echarlie**
supercedes wuvt/wuvt-site#385

fixes #12 
partial #15

### CODE REVIEW

File: donormotor/templates/admin/donation_index.html [hunk 1]
```diff
@@ -62,7 +67,12 @@ <h1>Donations</h1>
                 {{ donation.city }}, {{ donation.state }} {{ donation.zipcode }}
 {% endif %}
             </td>
-            <td>{% if donation.shipped_date %}{{ donation.shipped_date|datetime }} {% endif %}</td>
+            <td>{% if donation.shipped_date %}{{ donation.shipped_date|datetime }}
+		{% else %}{% if donation.premiums is not none %} <form method="post">
```
Comment (echarlie) [inline donormotor/templates/admin/donation_index.html]: I think elif is part of jinja2. I need to check this.

File: donormotor/admin/views.py [hunk 2]
```diff
@@ -67,6 +67,28 @@ def donation_index():
...
+            # TODO log who set as paid
+            o = Order.query.get(request.form["id"])
+            o.set_paid(method="check")
+            try:
+                db.session.commit()
+            except:
+                db.session.rollback()
+                raise
+
+        # TODO support setting asynchronously, rather than redirecting
```
Comment (echarlie) [inline donormotor/admin/views.py:80]: Create an issue for this.

File: donormotor/templates/admin/donation_index.html [hunk 3]
```diff
@@ -51,7 +51,12 @@ <h1>Donations</h1>
             <td>{{ donation.email }}</td>
             <td>{{ donation.phone }}</td>
             <td>{{ donation.placed_date|datetime }}</td>
-            <td>{% if donation.paid_date %}{{ donation.paid_date|datetime }}{% endif %}</td>
+	    <td>{% if donation.paid_date %}{{ donation.paid_date|datetime }}
```
Comment (echarlie) [inline donormotor/templates/admin/donation_index.html]: There may be an issue here with unpaid donations generated with cardtesting. These should not allow setting paid. c.f. #9

Comment (echarlie) [reply donormotor/templates/admin/donation_index.html]: also, whitespace needs fixed here.

**Review by echarlie — COMMENTED** [review]
I apparently can't "request changes" on my own review. But I made some notes so that I look at these specific issues.

**Review by JasonDavidoff85 — APPROVED** [review]
didnt launch it myself but seems good

**Review by mutantmonkey — APPROVED** [review]

**Pull request merged**